tests/RollbarTest.php

```diff
@@ -251,4 +251,23 @@ public function testEnable()
         Rollbar::configure(array('enabled' => false));
         $this->assertTrue(Rollbar::disabled());
     }
+
+    public function testLogUncaughtUnsetLogger()
+    {
+        $sut = new Rollbar;
+        $result = $sut->logUncaught('level', new \Exception);
+        $expected = new Response(0, "Rollbar Not Initialized");
+        $this->assertEquals($expected, $result);
+    }
+
+    public function testLogUncaught()
+    {
+        $sut = new Rollbar;
+        Rollbar::init(self::$simpleConfig);
+        $logger = Rollbar::logger();
+        $toLog = new \Exception;
+        $result = Rollbar::logUncaught(Level::ERROR, $toLog);
+        $this->assertEquals(200, $result->getStatus());
+        $this->assertObjectNotHasAttribute('uncaught', $toLog);
+    }
```

Comment:
`uncaught` is unset before and after invoking `logUncaught()`

The following assertion
```php
$this->assertObjectNotHasAttribute('uncaught', $toLog);
```
verifies the state after invocation but it should also verify the assumption of the unset variable before calling `logUncaught()` and verify that the state has changed inside the function.

Reply:
I considered adding that test, @bxsx, but opted not to because `logUncaught` does _not_ have a pre-condition on the presence or value of `$toLog::uncaught`. However, `logUncaught` _does_ have a _post-condition_ that `$toLog::uncaught` is unset.

That said, we should refactor all of this in the future: there is a [proposal to deprecate dynamic properties on objects in 8.2][1] (on which I voted "yes") that will require a different approach should it pass.

[1]:https://wiki.php.net/rfc/deprecate_dynamic_properties

Reply:
Fair point! Thanks for the elaboration.